x/sysproxy/sysproxy_linux.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package sysproxy
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+type ProxyType string
+
+const (
+	httpProxyType  ProxyType = "http"
+	httpsProxyType ProxyType = "https"
+	ftpProxyType   ProxyType = "ftp"
+)
+
+func SetProxy(ip string, port string) error {
+	// Execute Linux specific commands to set proxy
+	if err := setManualMode(); err != nil {
+		return err
+	}
+	if err := setWebProxy(httpProxyType, ip, port); err != nil {
+		return err
+	}
+	if err := setWebProxy(httpsProxyType, ip, port); err != nil {
+		return err
+	}
+	if err := setWebProxy(ftpProxyType, ip, port); err != nil {
+		return err
+	}
+	return nil
+}
+
+func setManualMode() error {
+	return execCommand("gsettings", "set", "org.gnome.system.proxy", "mode", "manual")
+}
+
+func setWebProxy(pType ProxyType, ip string, port string) error {
```

Comment:
You shouldn't have the type in the parameter and function name.
```suggestion
func setProxy(pType ProxyType, ip string, port string) error {
```